packages/vite-plugin-svelte/src/plugins/compile.js

```diff
@@ -0,0 +1,166 @@
+import { toRollupError } from '../utils/error.js';
+import { log, logCompilerWarnings } from '../utils/log.js';
+import fs from 'node:fs';
+import { ensureWatchedFile } from '../utils/watch.js';
+
+/**
+ * @param {import('../types/plugin-api.d.ts').PluginAPI} api
+ * @returns {import('vite').Plugin}
+ */
+export function compile(api) {
+	/**
+	 * @type {import("../types/options.js").ResolvedOptions}
+	 */
+	let options;
+
+	/**
+	 * @type {import("../types/compile.d.ts").CompileSvelte}
+	 */
+	let compileSvelte;
+	/** @type {import('vite').Plugin} */
+	const plugin = {
+		name: 'vite-plugin-svelte:compile',
+		configResolved() {
+			//@ts-expect-error defined below but filter not in type
+			plugin.transform.filter = api.idFilter;
+			options = api.options;
+			compileSvelte = api.compileSvelte;
+		},
+		transform: {
+			async handler(code, id) {
+				const ssr = this.environment.config.consumer === 'server';
+				const svelteRequest = api.idParser(id, ssr);
+				if (!svelteRequest) {
+					return;
+				}
+				const cache = api.getEnvironmentCache(this);
+				let compileData;
+				try {
+					const svelteMeta = this.getModuleInfo(id)?.meta?.svelte;
+					compileData = await compileSvelte(svelteRequest, code, options, svelteMeta?.preprocessed);
+				} catch (e) {
+					cache.setError(svelteRequest, e);
+					throw toRollupError(e, options);
+				}
+				if (compileData.compiled?.warnings) {
+					logCompilerWarnings(svelteRequest, compileData.compiled.warnings, options);
+				}
+				if (svelteRequest.raw) {
```

Comment:
this block of code is for an experimental feature returning data for import queries like `App.svelte?raw&svelte&type=script`, however that feature has remained unused for a long time.

with the split between preprocess & compile it got more complicated to support it so i propose to remove the feature entirely and only keep basic "raw" support from vite that returns the non-transformed file.